Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -50,42 +50,56 @@ def edit
         [HOMEBREW_REPOSITORY]
       end
     else
+      edit_api_message_displayed = T.let(false, T::Boolean)
       args.named.to_paths.select do |path|
-        next path if path.exist?
+        core_formula_path = path.fnmatch?("**/homebrew-core/Formula/*.rb", File::FNM_DOTMATCH)
+        core_cask_path = path.fnmatch?("**/homebrew-cask/Casks/*.rb", File::FNM_DOTMATCH)
+        core_formula_tap = path == CoreTap.instance.path
+        core_cask_tap = path == CoreCaskTap.instance.path
 
-        not_exist_message = if args.cask?
-          "#{path.basename(".rb")} doesn't exist on disk."
-        else
-          "#{path} doesn't exist on disk."
+        if path.exist?
+          if (core_formula_path || core_cask_path || core_formula_tap || core_cask_tap) \
+                && !edit_api_message_displayed \
+                && !Homebrew::EnvConfig.no_install_from_api? \
+                && !Homebrew::EnvConfig.no_env_hints?
+            opoo <<~EOS
+              Unless `HOMEBREW_NO_INSTALL_FROM_API` is set when running `brew install`,
+              it will ignore any locally edited #{(core_cask_path || core_cask_tap) ? "cask" : "formula"}.
+            EOS
+            edit_api_message_displayed = true
+          end
+          next path
         end
 
-        message = if args.cask?
-          <<~EOS
-            #{not_exist_message}
-            Run #{Formatter.identifier("brew create --cask --set-name #{path.basename(".rb")} $URL")} \
-            to create a new cask!
-          EOS
+        name = path.basename(".rb").to_s
+
+        if (tap_match = Regexp.new(HOMEBREW_TAP_DIR_REGEX.source + /$/.source).match(path.to_s))
```

Comment:
Identifying the tap from a path seems like something we can extract to a method somewhere.

Reply:
Any location in particular?